src/fpm_compiler.F90

```diff
@@ -1104,6 +1107,18 @@ subroutine compile_fortran(self, input, output, args, log_file, stat)
         & echo=self%echo, verbose=self%verbose, redirect=log_file, exitstat=stat)
 end subroutine compile_fortran
 
+subroutine generate_shared_library(self, package_name, output_dir)
+    !> Instance of the compiler object
+    class(compiler_t), intent(in) :: self
+    !> Name of the package.
+    character(len=*), intent(in) :: package_name
+    !> Output directory of object files.
+    character(len=*), intent(in) :: output_dir
+
+    call run(self%fc // " --shared " // output_dir// "/" // package_name // "/*.o" // " -o " // "lib" // &
```

Comment:
Maybe this could use the archive we are already generating. A wildcard for all object files would not be acceptable in any case, let's be explicit about them.